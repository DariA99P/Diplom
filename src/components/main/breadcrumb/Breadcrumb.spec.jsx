import React from 'react';
import { shallow } from 'enzyme';

import { BreadcrumbComponent } from '.';

jest.mock('constants/breadcrumbNameMap', () => ({
  '/home': 'Home',
  '/home/users': 'Users',
  '/home/users/my-account': 'My Account',
}));

describe('<Breadcrumb />', () => {
  describe('On rendering', () => {
    const location = { pathname: 'home/users/my-account' };

    it('should renders properly', () => {
      const wrapper = shallow(<BreadcrumbComponent location={location} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
