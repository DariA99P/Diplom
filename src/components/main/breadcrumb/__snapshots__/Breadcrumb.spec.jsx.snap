// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> On rendering should renders properly 1`] = `
<Breadcrumb
  separator="/"
  style={
    Object {
      "margin": "1rem 0",
    }
  }
>
  <BreadcrumbItem
    key="home"
    separator="/"
  >
    <Link
      href={true}
      replace={false}
      to="/"
    >
      Dashboard
    </Link>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="/home"
    separator="/"
  >
    <Link
      replace={false}
      to="/home"
    >
      Home
    </Link>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="/home/users"
    separator="/"
  >
    <Link
      replace={false}
      to="/home/users"
    >
      Users
    </Link>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="/home/users/my-account"
    separator="/"
  >
    <Link
      replace={false}
      to="/home/users/my-account"
    >
      My Account
    </Link>
  </BreadcrumbItem>
</Breadcrumb>
`;
